src/transformers/models/swin/modeling_tf_swin.py

```diff
@@ -222,13 +222,10 @@ def window_partition(input_feature: tf.Tensor, window_size: int) -> tf.Tensor:
     return windows
 
 
-def window_reverse(windows: tf.Tensor, window_size: int, height: int, width: int) -> tf.Tensor:
+def window_reverse(windows: tf.Tensor, batch_size: int, window_size: int, height: int, width: int) -> tf.Tensor:
     """
     Merges windows to produce higher resolution features.
     """
-    x = shape_list(windows)[0]
-    y = tf.cast(height * width / (window_size * window_size), tf.int32)
-    batch_size = int(x / y)
     windows = tf.reshape(
```

Comment:
I'd prefer it if we could keep the same signature for the pytorch and tensorflow module level functions. It's good you dug into this as it highlights a bug in my original implementation (using `int` here vs. `floor` in the pytorch version). 

What do you think about the following as an alternative?: 
```
def window_reverse(windows: tf.Tensor, window_size: int, height: int, width: int) -> tf.Tensor:
    """
    Merges windows to produce higher resolution features.
    """
    x = tf.shape(windows)[0]
    y = tf.cast(height * width / (window_size * window_size), tf.int32)
    batch_size = tf.math.floordiv(x, y)
    windows = tf.reshape(
        windows, (batch_size, height // window_size, width // window_size, window_size, window_size, -1)
    )
    windows = tf.transpose(windows, (0, 1, 3, 2, 4, 5))
    windows = tf.reshape(windows, (batch_size, height, width, -1))
    return windows
```

Reply:
Great! I'll apply this version to this PR.